pkg/agent/util/ipset/ipset_test.go

```diff
@@ -0,0 +1,199 @@
+			expectedArgs: []string{"ipset", "create", testIPSetName, string(HashNet), "-exist"},
+		},
+		{
+			name:         "Create IPv6 hash:ip set",
+			setType:      HashIP,
+			isIPv6:       true,
+			expectedArgs: []string{"ipset", "create", testIPSetName, string(HashIP), "family", "inet6", "-exist"},
+		},
+		{
+			name:              "Create IPv4 set with error",
+			setType:           HashIPPort,
+			err:               errors.New("some errors"),
+			output:            "error output",
+			expectedArgs:      []string{"ipset", "create", testIPSetName, string(HashIPPort), "-exist"},
+			expectedErrOutput: fmt.Sprintf("error creating ipset %s, err: some errors, output: error output", testIPSetName),
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			fcmd := exectesting.FakeCmd{
+				CombinedOutputScript: []exectesting.FakeAction{func() ([]byte, []byte, error) { return []byte(tt.output), nil, tt.err }},
+			}
+			fakeExec := &exectesting.FakeExec{
+				CommandScript: []exectesting.FakeCommandAction{
+					func(cmd string, args ...string) exec.Cmd { return exectesting.InitFakeCmd(&fcmd, cmd, args...) },
+				},
+			}
+			c := &Client{exec: fakeExec}
+			err := c.CreateIPSet(testIPSetName, tt.setType, tt.isIPv6)
+			assert.Equal(t, 1, fcmd.CombinedOutputCalls)
+			assert.Equal(t, tt.expectedArgs, fcmd.CombinedOutputLog[0])
+			if tt.expectedErrOutput != "" {
+				assert.EqualError(t, err, tt.expectedErrOutput)
+			} else {
+				assert.NoError(t, err)
+			}
+		})
+	}
+}
+
+func TestDestroyIPSet(t *testing.T) {
+	tests := []struct {
+		name              string
+		err               error
+		output            string
+		expectedErrOutput string
+	}{
+		{
+			name: "Destroy set successfully",
+		},
+		{
+			name: "Destroy non-existent set, no error",
+			err:  errors.New("The set with the given name does not exist"),
+		},
+		{
+			name:              "Destroy set with other error",
+			err:               errors.New("some errors"),
+			output:            "error output",
+			expectedErrOutput: fmt.Sprintf("error destroying ipset %s, err: some errors, output: error output", testIPSetName),
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			fcmd := exectesting.FakeCmd{
+				CombinedOutputScript: []exectesting.FakeAction{func() ([]byte, []byte, error) { return []byte(tt.output), nil, tt.err }},
+			}
+			fakeExec := &exectesting.FakeExec{
+				CommandScript: []exectesting.FakeCommandAction{
+					func(cmd string, args ...string) exec.Cmd { return exectesting.InitFakeCmd(&fcmd, cmd, args...) },
+				},
+			}
+			c := &Client{exec: fakeExec}
+			err := c.DestroyIPSet(testIPSetName)
+			assert.Equal(t, 1, fcmd.CombinedOutputCalls)
+			assert.Equal(t, []string{"ipset", "destroy", testIPSetName}, fcmd.CombinedOutputLog[0])
+			if tt.expectedErrOutput != "" {
+				assert.EqualError(t, err, tt.expectedErrOutput)
+			} else {
+				assert.NoError(t, err)
+			}
+		})
+	}
+}
+
+func TestAddEntry(t *testing.T) {
```

Comment:
No a big deal, but I am curious about why you didn't use table-driven test cases for these 3 test functions (`TestAddEntry`, `TestDelEntry`, `TestListEntries`), while you did for the other 2 (`TestCreateIPSet`, `TestDestroyIPSet`)?

Reply:
Nothing special. Just because I could write less code. I have updated these 3 ones to table driven style.